### ISSUE TITLE
[Accessibility]: Search homepage screen reader issues

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34735]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34189

### Problem description

The following screen reader navigation issues are present on the search homepage:

- [ ] Clicking "Show more" buttons in panels doesn't announce anything and keeps focus on the "show more" button (should focus on the newly added item instead)
- [ ] Repositories panel, recent files panel, and saved searches panel contents should be tables with headings, not a list (and make sure they don't read "bullet" before each item anymore)
- [ ] "My searches"/"All searches" should be exposed as tabs, not buttons

### Expected behavior

Fixed as described above

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
